### DISCUSSION TITLE
Install multiple PHP versions on the same system

### DISCUSSION THREAD

**Discussion by hakre** [shivammathur/setup-php#736] [Q&A / Help]
I'd like to run a matrix build against multiple PHP versions on the same virtual machine.

For example, the commands are named `php7.4`, `php8.0`, `php8.1` and `php8.2` as when running on `ubuntu 20.04` just on `ubuntu 22.04` or on `ubuntu 20.04` for the other php commands named `php5.3`, `php5.4`, `php5.5`, `php5.6`, `php7.0`, `php7.1`, `php7.2`, and `php7.3`.

Is there a way to use `shivammathur/setup-php` this way?

**Comment by hakre** [comment]
There is [`shivamathur/spc` on Packagist](https://packagist.org/packages/shivammathur/spc) which can narrow the gap on `ununtu 22.04` already. Tested with `php5.3` ... `php8.3`.

| #  | NOK    | OK     |
|----|--------|--------|
| 1  | php5.3 | php5.6 |
| 2  | php5.4 | php7.0 |
| 3  | php5.5 | php7.1 |
| 4  |        | php7.2 |
| 5  |        | php7.3 |
| 6  |        | php7.4 |
| 7  |        | php8.0 |
| 8  |        | php8.1 |
| 9  |        | php8.2 |
| 10 |        | php8.3 |

_(this is documented behavior.)_

There is also [`shivammathur/php5-ubuntu`](https://github.com/shivammathur/php5-ubuntu) which allows to install PHP 5.3 - 5.5.

This rudimentary setup takes roughly 5 minutes running on `ubuntu 22.04` and roughly 4 minutes running `ubuntu 20.04`.

**Comment by shivammathur** [comment]
GitHub Actions has no syntax to loop matrix variable for a single action and setup-php does not support installing multiple PHP versions in a single run.

You can run setup-php as one step after the other with different PHP versions in a workflow or use spc in a for loop.

**Reply by hakre** [reply]
Thank you for the feedback. Right now I distribute the installation across both installers, have not done yet much with extensions, so far this is stable and allows a little bit of parallel workload:

~~~bash
make -f .github/workflows/Makefile -O -j 2 setup
make -f .github/workflows/Makefile -O -j 2 phpLanguageLevels="5.3 5.6"
make -f .github/workflows/Makefile -O -j 2 phpLanguageLevels="5.4 7.0"
make -f .github/workflows/Makefile -O -j 2 phpLanguageLevels="5.5 7.1"
make -f .github/workflows/Makefile         phpLanguageLevels="7.2 7.3 7.4 8.0 8.1 8.2 8.3"
~~~
